### PR TITLE
Htysh/if 4274 support connection azure

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: fic/provider_test.go [hunk 1]
```diff
@@ -91,6 +93,17 @@ func testAccPreCheckRouterToECLConnection(t *testing.T) {
 	}
 }
 
+func testAccPreCheckAzureConnection(t *testing.T) {
+	testAccPreCheckRequiredEnvVars(t)
+
+	if OS_AZURE_SERVICE_KEY == "" {
```
Comment (keiichi-hikita) [inline fic/provider_test.go:101]: Please make either one of following change.

- If you want to skip when env var is not set, please change message like "This environment only runs ... tests"
- If you want to make tests fail, change method from Skip to Fatal

File: fic/provider.go [hunk 2]
```diff
@@ -180,16 +180,20 @@ func Provider() terraform.ResourceProvider {
...
+			"fic_eri_nat_component_v1":                        resourceEriNATComponentV1(),
+			"fic_eri_nat_global_ip_address_set_v1":            resourceEriNATGlobalIPAddressSetV1(),
+			"fic_eri_port_to_port_connection_v1":              resourceEriPortToPortConnectionV1(),
+			"fic_eri_port_v1":                                 resourceEriPortV1(),
+			"fic_eri_router_to_ecl_connection_v1":             resourceEriRouterToECLConnectionV1(),
+			"fic_eri_router_paired_to_port_connection_v1":     resourceEriRouterPairedToPortConnectionV1(),
+			"fic_eri_router_single_to_port_connection_v1":     resourceEriRouterSingleToPortConnectionV1(),
+			"fic_eri_router_to_uno_connection_v1":             resourceEriRouterToUNOConnectionV1(),
+			"fic_eri_router_v1":                               resourceEriRouterV1(),
+			"fic_eri_port_to_azure_microsoft_connection_v1":   resourceEriPortToAzureMicrosoftConnectionV1(),
```
Comment (keiichi-hikita) [inline fic/provider.go]: How about change the order as alpha-numeric, like ECL provider.

File: fic/resource_eri_port_to_azure_private_connection_v1.go [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+				ForceNew: true,
+			},
+
+			"source_secondary_vlan": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"source_asn": &schema.Schema{
```
Comment (keiichi-hikita) [inline fic/resource_eri_port_to_azure_private_connection_v1.go:60]: Could it be that API specification of FIC is wrong?
In API specification, ASN is defined as number format.
But this parameter is defines as string both in FIC provider and go-fic.

File: website/docs/r/eri_router_to_azure_microsoft_connection_v1.html.markdown [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    name = "terraform_connection_1"
+
+    source_router_id = "${fic_eri_router_v1.router_1.id}"
+    source_group_name = "group_1"
+    source_route_filter_in = "noRoute"
+    source_route_filter_out = "noRoute"
+
+    destination_interconnect = "Osaka-1"
+    destination_qos_type = "guarantee"
+    destination_service_key = "%s"
```
Comment (keiichi-hikita) [inline website/docs/r/eri_router_to_azure_microsoft_connection_v1.html.markdown]: "%s" is golang's format.
Please replace this into some string.

**Review by keiichi-hikita — CHANGES_REQUESTED** [review]

File: fic/resource_eri_port_to_azure_private_connection_v1.go (hunk 3 above)
Comment (ghost) [reply fic/resource_eri_port_to_azure_private_connection_v1.go:60]: To accordance with the API document:

https://fic.ntt.com/documents/api-references/connection-azure/rsts/Port%20to%20Azure(MS).html

, actual type of the parameter is string. The test code is successfully done with it in fact.
Therefore, I think this is an document bug:|

File: fic/provider_test.go (hunk 1 above)
Comment (ghost) [reply fic/provider_test.go:101]: Fixed!

File: fic/provider.go (hunk 2 above)
Comment (ghost) [reply fic/provider.go]: Fixed!

File: website/docs/r/eri_router_to_azure_microsoft_connection_v1.html.markdown (hunk 4 above)
Comment (ghost) [reply website/docs/r/eri_router_to_azure_microsoft_connection_v1.html.markdown]: Fixed!

**Review by keiichi-hikita — APPROVED** [review]
LGTM o/

**Review by shimisho — APPROVED** [review]
LGTM.

**Review by keiichi-hikita — APPROVED** [review]
LGTM o/

**Pull request merged**